Sources/Datadog/Tracing/DDSpanContext.swift

```diff
@@ -5,6 +5,17 @@
  */
 
 import Foundation
+import os.activity
+
+// Bridging Obj-C variabled defined as c-macroses. See `activity.h` header.
+private
+let OS_ACTIVITY_CURRENT = unsafeBitCast( dlsym( UnsafeMutableRawPointer(bitPattern: -2), "_os_activity_current"), to: os_activity_t.self)
+
+@_silgen_name("_os_activity_create") private
+func _os_activity_create(_ dso: UnsafeRawPointer?,
+                         _ description: UnsafePointer<Int8>,
+                         _ parent: Unmanaged<AnyObject>?,
+                         _ flags: os_activity_flag_t) -> AnyObject!
```

Comment:
i can't find `activity.h`, can we please add some more comments about this part? why we need it, how it works, etc.?
maybe we can comment the source of this code for more context as well

Reply:
Activity tracing is not very much documented, but it allows to keep the context trought the different threads, inheriting parent context when a thread is created, or when some part of code is executed in a separated queue, it keeps the activity that was active when the code is run. 
We map this activities with spans, so we can know which span was active when another was started, even in different queues ( there is a test `testActiveSpanIsKeptPerTask()` where this behavious is shown. In this way you can also know when a network requests return which was the active span that launched the request, not the span that was recently started.